pyeffects/Future.py

```diff
@@ -198,5 +198,33 @@ def on_complete(self, subscriber: Callable[[A], None]) -> None:
             self.subscribers.append(subscriber)
             self.semaphore.release()
 
+    def on_success(self, subscriber: Callable[[A], None]) -> None:
+        """Calls a subscriber function when :class:`Future <Future>` completes successfully.
+
+        :param subscriber: function to call when :class:`Future` completes successfully.
+
+        Usage::
+
+          >>> from pyeffects.Future import *
+          >>> val = Future.of(5).map(lambda v: v * v)
+          >>> val.on_success(lambda v: print(v))
+          Success(25)
+
+          >>> def error():
+          ...   raise RuntimeError()
+          >>> Future.run(error).on_success(lambda _: print(42))
+        """
+        self.semaphore.acquire()
+        if self.is_failure():
+            self.semaphore.release()
```

Comment:
one more comment, this block isn't currently tested.  

might want to add one more test, where you run an unsuccessful future and then run `on_complete`.